stubservice/stubhandlers/utils.go

```diff
@@ -95,12 +100,32 @@ type modifyStubError struct {
 	Code string
 }
 
-func modifyStub(st *stub, attributionCode string) (res *stub, err error) {
+func modifyStub(st *stub, attributionCode string, os string) (res *stub, err error) {
 	metrics.Statsd.Increment("modify_stub")
 
 	body := st.body
 	if attributionCode != "" {
-		if body, err = stubmodify.WriteAttributionCode(st.body, []byte(attributionCode)); err != nil {
+		switch os {
+		case "win":
+			// Windows exe attribution
+			if body, err = stubmodify.WriteAttributionCode(st.body, []byte(attributionCode)); err != nil {
+				return nil, &modifyStubError{err, attributionCode}
+			}
+		case "osx":
+			// Mac DMG attribution
+			dmgbody, er := dmglib.ParseDMG(bytes.NewReader(body))
+			if er != nil {
+				// Error parsing the DMG
+				return nil, &modifyStubError{err, attributionCode}
+			}
+			// Update the body in-place
+			// TODO: Maybe WriteAttributionCode should own sentinel value
```

Comment:
Yeah, probably it should. For Windows, it's defined here: https://github.com/mozilla-services/stubattribution/blob/e2fa0c91f6eb7df70dfdb6378b2bb76c5ac7fa8d/stubmodify/modify.go#L12